backend/src/main/java/com/sep6/backend/controller/AccountsController.java

```diff
@@ -17,28 +22,94 @@ public class AccountsController {
     private AccountsService service;
 
     @PutMapping(value = "{id}")
-    public void editAccount(@PathVariable int id, @RequestBody Account account){
-        service.editAccount(id, account);
+    public ResponseEntity<Account> editAccount(@PathVariable int id, @RequestBody Account account)
+    {
+        try
+        {
+            return ResponseEntity.ok(service.editAccount(id, account));
+        }
+        catch (NoSuchElementException e)
+        {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Account with id " + id + " does not exist.", e);
+        }
+        catch (Exception e)
+        {
+            //TODO log the error
+            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Something went wrong, please try again later");
+        }
```

Comment:
You could have just assign the NoSuchElementException a status code without having to rethrow it again